wrapanapi/systems/msazure.py

```diff
@@ -661,6 +661,14 @@ def list_free_pip(self, pip_template, resource_group=None):
         ips = self.network_client.public_ip_addresses.list(resource_group_name=resource_group)
         return [ip.name for ip in ips if not ip.ip_configuration and pip_template in ip.name]
 
+    def list_all_free_pips(self, resource_group=None):
+        """
+        List all available Public IPs in the selected resource_group
+        """
+        resource_group = resource_group or self.resource_group
+        ips = self.network_client.public_ip_addresses.list(resource_group_name=resource_group)
+        return [ip.name for ip in ips if not ip.ip_address]
```

Comment:
I think it would be better to return the instances, instead of only their names. Let the caller process the names out.

Reply:
Expanded the diff some more - same comment about the pattern in the functions applies here. This should be a mode of operation for the existing `list_free_pip` function, which would have a `pip_template=None` default.